src/utils/preview-service.ts

```diff
@@ -19,8 +19,21 @@ export type ChainConfig = {
   etherscanUrl: string;
 };
 
+/**
+ * @dev pings the metal service to ensure it is running
+ */
+export const pingMetalService = async () => {
+  await fetch(`${METAL_SERVICE_URL}`).catch(async e => {
+    logDebug(e);
+    await exit(
+      'Error! Cannot connect to Metal servers',
+      MODE === 'dev' ? 'Make sure metal-service is running on port 1234' : '',
+    );
+  });
+};
```

Comment:
nice! it might also make sense to skip this fetch if NO_METAL_SERVICE is set.

Reply:
that's a really great note